platform/engineering/design-docs/2021-01-13-web-component-library.md

```diff
@@ -0,0 +1,139 @@
+
+#### Risks
+
+Despite polyfills, there is some risk of certain Web Component features not working properly in all browsers (IE 11, older versions of Edge). Due to existing polyfill support combined with the trend of users switching to more modern browsers over time, this is a low risk. If we encounter this problem, the workaround will be to avoid using whatever feature is under consideration and instead implement a solution that will yield the same results.
+
+### Work Estimates
+
+Note: These estimates do not account for time spent on other team duties such as support requests.
+
+#### Stencil Project setup with Storybook integration
+
+**Estimate:** ~3 days
+
+This has already been done as a PoC - some additional work should be done to make sure that the built version of storybook can use the Web Components
+
+#### Create new Webpack entry for Web Component loader
+
+**Estimate:** ~2 days
+
+`vets-website` should create a `wc-loader.entry.js` file that will get loaded in the `<head>` of each page.
+
+#### Develop an Accordion Web Component
+
+**Estimate**: ~4 days
+
+This will be the first WC to go live on va.gov.
+
+### Alternatives
+
+There isn't another way of achieving a proper (i.e., not copy-pasted HTML) component library that isn't specific to a single framework like React. However, there are other tools/frameworks besudes Stencil that we could use to build Web Components.
+
+There are [dozens of tools/frameworks](https://webcomponents.dev/blog/all-the-ways-to-make-a-web-component/) that can be used to build Web Components, with [LitElement](https://lit-element.polymer-project.org/) being one of the bigger ones. Here are some reasons that Stencil was chosen over all of these alternatives:
+
+- JSX support
+  - Stencil is one of the few tools for building Web Components that lets the developer use JSX  syntax, as opposed to a template string or direct DOM manipulation in order to create the HTML  nodes. This will be very familiar for React developers.
+- Typescript
+  - The Component source will be written in TypeScript, and static typing can help eliminate bugs in code as well as allow developers to gain a clearer understanding.
+- Auto-generated component docs
+  - Stencil is able to automatically generate a README for each component with documentation on its props, events, CSS variables, and more. This can be combined with Storybook.
+
+### Future Work
+
+Continue adding Components until the React version of `component-library` ic completely replaced. Web Components can co-exist fine with React components, but it will be easier if we only have to maintain a single library.
+
+#### Templating
+
+The library could be extended to include Components focused on overall site layout in addition to the more granular Components. This could allow us to use Web Components as a replacement for the Liquid templates in `vets-website`.
```

Comment:
Maybe link to https://stenciljs.com/docs/static-site-generation?

I'd also recommend saying only that this is an _option_, not a thing we will definitely do as future work.